tests/components/melnor/test_config_flow.py

```diff
@@ -11,6 +15,53 @@
 INTEGRATION_DISCOVERY = {CONF_MAC: FAKE_ADDRESS}
 
 
+async def test_pick_device(hass):
+    """Test we properly handle device picking."""
+
+    with patch(
+        "homeassistant.components.melnor.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry, patch(
+        "homeassistant.components.bluetooth.async_discovered_service_info",
+    ) as mock_async_discovered_service_info:
+
+        device = BluetoothServiceInfoBleak(
+            name="None",
+            address=FAKE_ADDRESS,
+            rssi=-19,
+            manufacturer_data={13: bytearray([89])},
+            service_data={},
+            service_uuids=[],
+            source="local",
+            device=BLEDevice(FAKE_ADDRESS, None),
+            advertisement=AdvertisementData(local_name=""),
+            time=0,
+            connectable=True,
+        )
+
+        mock_async_discovered_service_info.return_value = [device]
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_USER},
+        )
+
+        assert result["type"] == FlowResultType.FORM
+        assert result["step_id"] == "pick_device"
+        assert result["data_schema"].__dict__["schema"][CONF_ADDRESS].container == {
```

Comment:
`result["data_schema"]({})[CONF_ADDRESS]`

You could do something like this

Reply:
voluptuous gets unhappy when I do that. I'm not familiar enough with python to fully articulate the check here, but essentially I just want some simple equality check to validate the schema contains the found addresses.

```
FAILED tests/components/melnor/test_config_flow.py::test_pick_device[pyloop] - voluptuous.error.MultipleInvalid: required key not provided @ data['address']
```

Reply:
You could test it with
```
        result2 = await hass.config_entries.flow.async_configure(
            result["flow_id"], user_input={CONF_ADDRESS: "wrong address here"}
        )

       verify it rejects....then try again with a valid one

        result3 = await hass.config_entries.flow.async_configure(
            result2["flow_id"], user_input={CONF_ADDRESS: FAKE_ADDRESS}
        )
```